src/content/configuration/output.mdx

```diff
@@ -1305,6 +1305,206 @@ W> Note that not setting a `output.library` will cause all properties returned b
+
+- #### `libraryTarget: 'amd-require'`
+
+  This packages your output with an immediately-executed AMD `require(dependencies, factory)` wrapper.
+
+  The `'amd-require'` target allows for the use of AMD dependencies without needing a separate later invocation. As with the `'amd'` target, this depends on the appropriate [`require` function](https://github.com/amdjs/amdjs-api/blob/master/require.md) being available in the environment in which the webpack output is loaded.
+
+  With this target, the library name is ignored.
+
+- #### `libraryTarget: 'umd'`
+
+  This exposes your library under all the module definitions, allowing it to work with CommonJS, AMD and as global variable. Take a look at the [UMD Repository](https://github.com/umdjs/umd) to learn more.
+
+  In this case, you need the `library` property to name your module:
+
+  ```javascript
+  module.exports = {
+    //...
+    output: {
+      library: 'MyLibrary',
+      libraryTarget: 'umd',
+    },
+  };
+  ```
+
+  And finally the output is:
+
+  ```javascript
+  (function webpackUniversalModuleDefinition(root, factory) {
+    if (typeof exports === 'object' && typeof module === 'object')
+      module.exports = factory();
+    else if (typeof define === 'function' && define.amd) define([], factory);
+    else if (typeof exports === 'object') exports['MyLibrary'] = factory();
+    else root['MyLibrary'] = factory();
+  })(typeof self !== 'undefined' ? self : this, function () {
+    return _entry_return_;
+  });
+  ```
+
+  Note that omitting `library` will result in the assignment of all properties returned by the entry point be assigned directly to the root object, as documented under the [object assignment section](#expose-via-object-assignment). Example:
+
+  ```javascript
+  module.exports = {
+    //...
+    output: {
+      libraryTarget: 'umd',
+    },
+  };
+  ```
+
+  The output will be:
+
+  ```javascript
+  (function webpackUniversalModuleDefinition(root, factory) {
+    if (typeof exports === 'object' && typeof module === 'object')
+      module.exports = factory();
+    else if (typeof define === 'function' && define.amd) define([], factory);
+    else {
+      var a = factory();
+      for (var i in a) (typeof exports === 'object' ? exports : root)[i] = a[i];
+    }
+  })(typeof self !== 'undefined' ? self : this, function () {
+    return _entry_return_;
+  });
+  ```
+
+  Since webpack 3.1.0, you may specify an object for `library` for differing names per targets:
+
+  ```javascript
+  module.exports = {
+    //...
+    output: {
+      library: {
+        root: 'MyLibrary',
+        amd: 'my-library',
+        commonjs: 'my-common-library',
+      },
+      libraryTarget: 'umd',
```

Comment:
It can be rewritten on `library.name = { root, amd, commonjs }`, we should avoid using `libraryTarget`

Reply:
`libraryTarget` only for compatibility, all library options should be in `output.library`

Reply:
All values of `libraryTarget` will have respective `library.type` item. For example, here's the new one for `umd` type https://github.com/webpack/webpack.js.org/pull/4691/files#diff-24a08fc6495329508b1a44d5f20bdaf472555d1c495157ee021f9405dea9bf5dR1056.

But I think we could still keep documentation for `libraryTarget` for a while so users can migrate. I'm going to inject a warning at the beginning of every `libraryTarget` encouraging users to use the respective `library.type`.

Reply:
Yep, just think we should provide all examples using `library`, and provide notes about `libraryTarget` with small example